osbs/core.py

```diff
@@ -320,21 +321,54 @@ def get_build_config(self, build_config_id):
         build_config = response.json()
         return build_config
 
+    def get_all_build_configs_by_labels(self, label_selectors):
+        """
+        Returns all builds matching a given set of label selectors. It is up to the
+        calling function to filter the results.
+        """
+        labels = ['%s=%s' % (field, value) for field, value in label_selectors]
+        labels = ','.join(labels)
+        url = self._build_url("buildconfigs/", labelSelector=labels)
+        return self._get(url).json()['items']
+
     def get_build_config_by_labels(self, label_selectors):
         """
         Returns a build config matching the given label
         selectors. This method will raise OsbsException
         if not exactly one build config is found.
         """
-        labels = ['%s=%s' % (field, value) for field, value in label_selectors]
-        labels = ','.join(labels)
-        url = self._build_url("buildconfigs/", labelSelector=labels)
-        items = self._get(url).json()['items']
+        items = self.get_all_build_configs_by_labels(label_selectors)
+
+        if not items:
+            raise OsbsException(
+                "Build config not found for labels: %r" %
+                (label_selectors, ))
+        if len(items) > 1:
+            raise OsbsException(
+                "More than one build config found for labels: %r" %
+                (label_selectors, ))
+
+        return items[0]
+
+    def get_build_config_by_labels_filtered(self, label_data):
+        """
+        Returns a build config matching the given label selectors, filtering against
+        another predetermined value. This method will raise OsbsException
+        if not exactly one build config is found.
+        """
+        label_selectors = label_data['search_labels']
+        items = self.get_all_build_configs_by_labels(label_selectors)
+        match_filter = label_data.get(FILTER_KEY) or None
 
         if not items:
             raise OsbsException(
                 "Build config not found for labels: %r" %
                 (label_selectors, ))
+        if match_filter is not None:
+            for build_config in items:
+                filter_value = graceful_chain_get(build_config, *FILTER_KEY.split('.'))
+                if filter_value == match_filter:
+                    return build_config
         if len(items) > 1:
```

Comment:
this is missing coverage, also 364, and 377

Reply:
Let's make sure this is taken care of before merging.

Reply:
Test coverage added.